### PR TITLE
fix(get dates to investigate error handling):

### PR DESCRIPTION
**Pull request by daniel7764**
1093

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 1]
```diff
@@ -167,8 +167,10 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
     }, []);
 
     useEffect(() => {
-        const datesToInvestigate = getDatesToInvestigate(doesHaveSymptoms,symptomsStartDate,coronaTestDate);
-        datesToInvestigate.length > 0 ? setDatesToInvestigate(datesToInvestigate) : alertError('תאריך תסמינים לא חוקי');
+        if(Boolean(coronaTestDate)) {
```
Comment (OrelKaravani) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:170]: It's not necessary.

File: client/src/Utils/DateUtils/useDateUtils.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
+import Swal from "sweetalert2";
 import {differenceInDays, eachDayOfInterval, subDays} from 'date-fns';
 
 export const symptomsWithKnownStartDate: number = 4;
 export const nonSymptomaticPatient: number = 7;
-export const symptomsWithUnknownStartDate: number = 7;
+export const symptomsWithUnknownStartDate: number = 10;
```
Comment (OrelKaravani) [inline client/src/Utils/DateUtils/useDateUtils.ts:6]: Why there is a change in the number of days?

File: client/src/Utils/DateUtils/useDateUtils.ts [hunk 3]
```diff
@@ -15,7 +17,11 @@ export const getDatesToInvestigate = (doesHaveSymptoms: boolean, symptomsStartDa
             if (symptomsStartDate) {
                 const testAndSymptomsInterval = Math.abs(differenceInDays(symptomsStartDate, coronaTestDate));
                 if (testAndSymptomsInterval > maxInvestigatedDays) {
-                    return []
+                    Swal.fire({
```
Comment (ItayBenMoshe) [inline client/src/Utils/DateUtils/useDateUtils.ts:20]: use our custom swal

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

File: client/src/Utils/DateUtils/useDateUtils.ts [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 
 export const symptomsWithKnownStartDate: number = 4;
 export const nonSymptomaticPatient: number = 7;
-export const symptomsWithUnknownStartDate: number = 7;
+export const symptomsWithUnknownStartDate: number = 10;
 const maxInvestigatedDays: number = 21;
 
 export const convertDate = (dbDate: Date | null) => dbDate ? new Date(dbDate) : null;
 
+
```
Comment (ItayBenMoshe) [inline client/src/Utils/DateUtils/useDateUtils.ts:11]: delete row

File: client/src/Utils/DateUtils/useDateUtils.ts [hunk 5]
```diff
@@ -1,12 +1,14 @@
+import Swal from "sweetalert2";
```
Comment (ItayBenMoshe) [inline client/src/Utils/DateUtils/useDateUtils.ts:1]: delete this import when using the custom swal

File: client/src/Utils/DateUtils/useDateUtils.ts (hunk 3 above)
Comment (OrelKaravani) [reply client/src/Utils/DateUtils/useDateUtils.ts:20]: It cannot be used out of a react component.

Comment (ItayBenMoshe) [reply client/src/Utils/DateUtils/useDateUtils.ts:20]: why not?

Comment (ItayBenMoshe) [reply client/src/Utils/DateUtils/useDateUtils.ts:20]: there is a lot of examples for this

Comment (daniel7764) [reply client/src/Utils/DateUtils/useDateUtils.ts:20]: The other solution is to have the function recive a callback function and sending it our custom swal, i don't think it's a better solution, maybe you have an idea?

Comment (OrelKaravani) [reply client/src/Utils/DateUtils/useDateUtils.ts:20]: > there is a lot of examples for this

I would like you to show me if it's possible. We tried.

Comment (ItayBenMoshe) [reply client/src/Utils/DateUtils/useDateUtils.ts:20]: use the alertError from the custom hook

Comment (ItayBenMoshe) [reply client/src/Utils/DateUtils/useDateUtils.ts:20]: you can see an example in useGroupedInvestigations

**Review by doronlnd84 — APPROVED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Pull request merged**